### PR TITLE
CLE-NST-A - Add support of extra parameters in macros

### PR DESCRIPTION
**Pull request by IgorWolkov**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -8,12 +8,12 @@ lazy val root = project
   .settings(testFrameworks += new TestFramework("munit.Framework"))
   .settings(Test / parallelExecution := true)
 
-lazy val dependencies = cats ++ circe ++ munit
+lazy val dependencies = circe ++ munit
```
Comment (Vladyslav-WConsulting) [inline build.sbt:11]: Why is the cats library removed from dependencies?

File: src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala [hunk 2]
```diff
@@ -85,7 +89,9 @@ object macros:
 
       import quotes.reflect._
 
-      val stringContextParts: List[String] = getStringContextParts(sc)
+      val stringContextParts: List[String] = getStringContextParts(sc) getStringContextParts(sc) match
```
Comment (mikhailZakharevich) [inline src/main/scala/group/scala/karazin/circe/literal/extras/macros.scala]: duplication?

**Review by mikhailZakharevich — CHANGES_REQUESTED** [review]

File: src/test/scala/group/scala/karazin/circe/literal/extras/suites/EitherEncodeSuite.scala [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+//              {
+//                "Right": "Hello world",
+//                "extraKey": "extraValue"
+//              }
+//              """"" + """"
+//        """
+//    ).headOption match
+//      case Some(error) => assert(error.message.startsWith("Encode error:"))
+//      case _           => fail("No compilation error was found.")
+//  }
```
Comment (VLADISLAV008) [inline src/test/scala/group/scala/karazin/circe/literal/extras/suites/EitherEncodeSuite.scala]: Why are these tests commented out?

Comment (IgorWolkov) [reply src/test/scala/group/scala/karazin/circe/literal/extras/suites/EitherEncodeSuite.scala]: Fixed.

File: build.sbt (hunk 1 above)
Comment (IgorWolkov) [reply build.sbt:11]: Remind me the necessity of it.

Comment (VLADISLAV008) [reply build.sbt:11]: Previously, dependencies did not have the package `cats-laws`. For this, `cats ` is separately added.
I did not notice that now the ` cats-laws` is separately added in the dependencies. Therefore, this comment is irrelevant.

**Review by VLADISLAV008 — APPROVED** [review]

**Pull request merged**